src/card_operations/card_fetch_challenge.h

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file    card_fetch_challenge.h
+ * @author  Cypherock X1 Team
+ * @brief   API for fetching wallet challenge from a card
+ * @copyright Copyright (c) 2023 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/"
+ * target=_blank>https://mitcc.org/</a>
+ */
+#ifndef CARD_FETCH_CHALLENGE_H
+#define CARD_FETCH_CHALLENGE_H
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "card_operation_typedefs.h"
+#include "flash_struct.h"
+#include "stdbool.h"
+#include "stdint.h"
+
+/*****************************************************************************
+ * MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * EXPORTED VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief Fetches wallet challenge data from a card.
+ * @details This function initializes the applet, retrieves the wallet challenge
+ * from the card. It handles various error cases and returns an appropriate
+ * error code. NOTE: The instruction to tap card is not displayed by the
+ * function as the api might be called as a continuation to error handling of a
+ * separate flow
```

Comment:
Do mention that the function internally updates flash data (such wallet lock/unlock state, challenge, etc.)

Reply:
added